src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    void M1<T>(T arg) { }
+    void M2()
+    {
+        [|M1|](1, 2);
+    }
+}",
+@"
+class C1
+{
+    void M1<T>(T arg, int v) { }
+    void M2()
+    {
+        M1(1, 2);
+    }
+}");
+            //Should fix to: void M1<T>(T arg, T v) { }
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationRecursion()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    void M1()
+    {
+        [|M1|](1);
+    }
+}",
+@"
+class C1
+{
+    void M1(int v)
+    {
+        M1(1);
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationOverloads1()
+        {
+            var code =
+@"
+class C1
+{
+    void M1(string s) { }
+    void M1(int i) { }
+    void M2()
+    {
+        [|M1|](1, 2);
+    }
+}";
+            var fix0 =
+@"
+class C1
+{
+    void M1(string s) { }
+    void M1(int i, int v) { }
+    void M2()
+    {
+        M1(1, 2);
+    }
+}";
+            var fix1 =
+@"
+class C1
+{
+    void M1(int v, string s) { }
+    void M1(int i) { }
+    void M2()
+    {
+        M1(1, 2);
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix0, 0);
+            await TestInRegularAndScriptAsync(code, fix1, 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationOverloads2()
+        {
+            var code =
+@"
+class C1
+{
+    void M1(string s1, string s2) { }
+    void M1(string s) { }
+    void M1(int i) { }
+    void M2()
+    {
+        M1(1, [|2|]);
+    }
+}";
+            var fix0 =
+@"
+class C1
+{
+    void M1(string s1, string s2) { }
+    void M1(string s) { }
+    void M1(int i, int v) { }
+    void M2()
+    {
+        M1(1, 2);
+    }
+}";
+            var fix1 =
+@"
+class C1
+{
+    void M1(string s1, string s2) { }
+    void M1(int v, string s) { }
+    void M1(int i) { }
+    void M2()
+    {
+        M1(1, 2);
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix0, 0);
+            await TestInRegularAndScriptAsync(code, fix1, 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationTuple1()
+        {
+            var code =
+@"
+class C1
+{
+    void M1((int, int) t1)
+    {
+    }
+    void M2()
+    {
+        [|M1|]((0, 0), (1, ""1""));
+    }
+}";
+            var fix0 =
+    @"
+class C1
+{
+    void M1((int, int) t1, (int, string) p)
```

Comment:
it would be nice if p is set to do rename - I know there is no reference to it since it just added. but since it is auto generated identifier, people probably will want to change it?

Reply:
tagging @jinujoseph @kuhlenh